source/common/http/http1/codec_impl.cc

```diff
@@ -788,8 +789,9 @@ Envoy::StatusOr<int> ConnectionImpl::onHeadersCompleteBase() {
 
   header_parsing_state_ = HeaderParsingState::Done;
 
-  // Returning 2 informs http_parser to not expect a body or further data on this connection.
-  return handling_upgrade_ ? 2 : statusor.value();
+  // Returning HttpParserCode::NoBodyData informs http_parser to not expect a body or further data
+  // on this connection.
+  return handling_upgrade_ ? HttpParserCode::NoBodyData : statusor;
```

Comment:
Nit: restore the use of "statusor.value()" in the else case.

Returning statusor.value() instead of statusor makes it clear that the statusor must not contiain an error status at this point.  Error status is handled further up.

Reply:
Done.